File: tot_spec/src/codegen/fixtures/typescript/enum_custom_tag.ts

export type Number =
        // Variant Int64
    { kind: "Int64", data: bigint }

    |     // Variant Float
    { kind: "Float", data: number }

    | { kind: "RealNumber", data: RealNumber }
;


export class RealNumber {
    part0: number | undefined;
    part1: number | undefined;

    constructor(data: Partial<RealNumber>) {
        Object.assign(this, data);
    }

    toJSON(): any {
        return {
            part_0: this.part0,
            part_1: this.part1,
        };
    }

    static fromJSON(json: any): RealNumber {
        return new RealNumber({
            part0: json.part_0,
            part1: json.part_1,
        });
    }
}

export type RealNumberJSON = {
    part_0: number | undefined;
    part_1: number | undefined;
}

File: tot_spec/src/codegen/fixtures/rs_serde/enum_custom_tag.rs
/// Number
#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
#[serde(tag = "kind", content = "data")]
pub enum Number {
    /// Variant Int64
    Int64(i64),
    /// Variant Float
    Float(f64),
    RealNumber(RealNumber),
}
/// RealNumber
#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
pub struct RealNumber {
    pub part_0: std::option::Option<f64>,
    pub part_1: std::option::Option<f64>,
}
